src/model/position.js

```diff
@@ -319,6 +319,30 @@ export default class Position {
 	}
 
 	/**
+	 * Returns a {@link module:engine/model/node~Node} which is a common ancestor for both positions. The {@link #root roots}
+	 * of these two positions must be identical.
+	 *
+	 * @param {module:engine/model/position~Position} position The second position.
+	 * @returns {module:engine/model/node~Node|null} Node that contains both positions.
+	 */
+	getCommonAncestor( position ) {
+		if ( this.root !== position.root ) {
+			return null;
+		}
+
+		const ancestorsA = this.getAncestors();
```

Comment:
I liked the previous implementation more. The one which reused `getCommonPath()`. It was shorter and more readable.

Let me guess – you decided to skip it because DF doesn't implement `getNodeByPath()`? ;) Why not borrowing it from `Element` for `DF`?